backend/src/common/util/workflow.go

```diff
@@ -51,11 +51,17 @@ func (w *Workflow) OverrideParameters(desiredParams map[string]string) {
 		if param, ok := desiredParams[currentParam.Name]; ok {
 			desiredValue = &param
 		} else {
-			desiredValue = currentParam.Value
+			val := currentParam.Value.String()
```

Comment:
nit: the `val` here is a different variable than the `val` below, shall we use different names?
can we improve this via `util.StringToPtr`?